### PR TITLE
feat(SubsurfaceViewer): Record-based definition of layers that supports TS Objects implemented.

### PR DESCRIPTION
**Pull request by LeonidPolukhin**
1) typedArraySupport optional flag is added to SubsurfaceViewer API. If set to true records with layer definitions are used for layer creation bypassing JSONConverter.
2) PointsLayer, TriangleLayer API are updated to accept typed arrays as input data skipping extra copying. The corresponding stories are added.
3) MapLayer and Grid3DLayer already support typed array inputs. The stories updated.

PolylinesLayer has an internal algo for making polylines closed when requested so its modification is postponed for a while.

Known issue: in the story book "typed array stories" don't work if any property of layer is changed in Controls Tab. It happens because the story book deserializes input arrays from JSON represented in Controls loosing typeinfos.

### CODE REVIEW

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx [hunk 1]
```diff
@@ -123,6 +123,11 @@ export interface SubsurfaceViewerProps {
     lights?: LightsType;
 
     children?: React.ReactNode;
+
+    /**
+     * Forces the component to use the records provided in 'layers' prop as is.
+     */
+    skipLayerPropsConversion?: boolean;
```
Comment (LeonidPolukhin) [inline typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx:130]: Not happy with this name, any better proposals?

File: typescript/packages/subsurface-viewer/src/layers/map/mapLayer.stories.tsx [hunk 2]
```diff
@@ -574,7 +574,9 @@ const TypedArrayInputComponent: React.FC<{
         id: "map",
         layers: [
             // Can not use Record<string, unknown> because makeData() is not supported
```
Comment (t0oF-azpn) [inline typescript/packages/subsurface-viewer/src/layers/map/mapLayer.stories.tsx:576]: Update the comment here

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx (hunk 1 above)
Comment (t0oF-azpn) [reply typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx:130]: jsObjectLayerPropsSupport ?

Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx:130]: or useTypedArrays? To underline the main purpose of the flag.

File: typescript/packages/subsurface-viewer/src/layers/map/mapLayer.stories.tsx (hunk 2 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/layers/map/mapLayer.stories.tsx]: Comment deleted.

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx (hunk 1 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/SubsurfaceViewer.tsx]: Renamed.

**Review by nilscb — APPROVED** [review]

**Pull request merged**